src/wallet/rpcwallet.cpp

```diff
@@ -2712,6 +2814,40 @@ UniValue z_listaddresses(const UniValue& params, bool fHelp)
     return ret;
 }
 
+UniValue z_listunifiedreceivers(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() < 1 || params.size() > 1)
+        throw runtime_error(
+            "z_listunifiedreceivers unified_address\n"
+            "\nReturns the (per-pool) receivers contained within the provided UA;"
+            "\nthe UA may not have receivers for some pools.\n"
+            "\nArguments:\n"
+            "1. unified_address (string) The unified addresses\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"transparent\": \"address\",   (string) The legacy transparent address\n"
+            "  \"sapling\": \"address\",       (string) The legacy sapling address\n"
+            "  \"orchard\": \"address\",       (string) The single-receiver unified address\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("z_listunifiedreceivers", "")
+            + HelpExampleRpc("z_listunifiedreceivers", "")
+        );
+
+    if (!fExperimentalOrchardWallet) {
+        throw JSONRPCError(RPC_WALLET_ENCRYPTION_FAILED, "Error: the Orchard wallet experimental extensions are disabled.");
+    }
+    std::string ua = params[0].get_str();
+    UniValue result(UniValue::VOBJ);
+    result.pushKV("transparent", "stubTransparantAddr");
+    result.pushKV("sapling", "stubSaplingAddr");
+    result.pushKV("orchard", "stubOrchardAddr " + ua);
```

Comment:
```suggestion
    result.pushKV("sapling", "TODO");
    result.pushKV("orchard", "TODO"); // this will be a unified address containing *only* the Orchard receiver.
```